test/src/specs/tx_pool/descendant.rs

```diff
@@ -0,0 +1,261 @@
+use crate::specs::tx_pool::utils::prepare_txfamily;
+use crate::{Net, Node, Spec};
+use ckb_types::core::{BlockView, TransactionView};
+use std::collections::HashSet;
+
+// Convention:
+//   * `tx_family` is a set of relative transactions, `tx_family.a <- tx_family.b <-
+//     tx_family.c <- tx_family.d <- tx_family.e`, note that `x <- y` represents `x` is the parent
+//     transaction of `y`.
+
+pub struct HandlingDescendantsOfProposed;
+
+impl Spec for HandlingDescendantsOfProposed {
+    crate::name!("handling_descendants_of_proposed");
+
+    // Case: This case intends to test the handling of proposed transactions.
+    //       We construct a scenario that although both `tx_family.a` and `tx_family.b` are in
+    //       txpool, but only propose `tx_family.a`. We expect that after proposing
+    //       `tx_family.a`, miner is able to propose `tx_family.b` in the next blocks.
+    fn run(&self, net: &mut Net) {
+        let node = &net.nodes[0];
+        let window = node.consensus().tx_proposal_window();
+        node.generate_blocks((window.farthest() + 2) as usize);
+
+        // 1. Put `tx_family` into pending-pool.
+        let family = prepare_txfamily(node);
+        node.submit_transaction(family.a());
+        node.submit_transaction(family.b());
+
+        // 2. Propose `tx_family.a` only, then we expect `tx_family.b` being proposed in the next
+        //    blocks, even after `tx_family.a` expiring, out of `tx_family.a`'s proposal window
+        node.submit_block(&propose(node, &[family.a()]));
+        (0..=window.farthest() + window.closest()).for_each(|_| {
+            let block = node.new_block(None, None, None);
+            assert!(
+                block
+                    .union_proposal_ids()
+                    .contains(&family.b().proposal_short_id()),
+                "Miner should propose tx_family.b since it has never been proposed, actual: {:?}",
+                block.union_proposal_ids(),
+            );
+
+            node.submit_block(&blank(node)); // continuously submit blank blocks.
+        });
+        (0..10).for_each(|_| {
```

Comment:
What's the `10` represents for?

Reply:
It is a meaningless number.
I have changed it. Please have a look.